src/Atom.jl

```diff
@@ -12,9 +12,28 @@ function __init__()
 
   atreplinit() do repl
     fixdisplayorder()
+    instantiate_repl_keybindings(repl)
   end
 
-  atreplinit(instantiate_repl_keybindings)
+  # HACK: overwrite Pkg mode `on_done` function after Pkg mode has been Initialized
+  f = (repl) -> begin
+    i = findfirst(Base.active_repl.interface.modes) do mode
+      mode isa LineEdit.Prompt &&
+      isdefined(mode, :on_done) &&
+      let ms = collect(methods(mode.on_done))
+        length(ms) > 0 && parentmodule(ms[1].module) == Pkg
+      end
+    end
+    i === nothing && return # if no Pkg mode somehow
+    pkg_mode = Base.active_repl.interface.modes[i]
+    pkg_mode.on_done = let original = pkg_mode.on_done
+      (s, buf, ok) -> begin
+        original(s, buf, ok)
+        update_project()
+      end
+    end
+  end
+  push!(Base.repl_hooks, f)
```

Comment:
unfortunately not, this overwriting only works after Pkg.jl's repl mode is created into `Base.active_repl.interface`, but since Pkg.jl is initialized before Atom.jl (and register its `atreplinit` hook before us), this function would be called before Pkg.jl's `atreplinit` hook if we just use `atreplinit`.

This is because `atreplinit` uses [`pushfirst`](https://github.com/JuliaLang/julia/blob/097f0b0c8612b4b172e403de2b852d962fb7bb6d/base/client.jl#L342) for registering a hook function.